processing/src/main/java/org/apache/druid/query/context/ResponseContext.java

```diff
@@ -76,56 +138,128 @@ public static ResponseContext createEmpty()
     return DefaultResponseContext.createEmpty();
   }
 
-  protected abstract Map<String, Object> getDelegate();
-
-  public Object put(String key, Object value)
+  public static ResponseContext deserialize(String responseContext, ObjectMapper objectMapper) throws IOException
   {
-    return getDelegate().put(key, value);
+    final Map<String, Object> delegate = objectMapper.readValue(
+        responseContext,
+        JacksonUtils.TYPE_REFERENCE_MAP_STRING_OBJECT
+    );
+    return new ResponseContext()
+    {
+      @Override
+      protected Map<String, Object> getDelegate()
+      {
+        return delegate;
+      }
+    };
   }
 
-  public Object get(String key)
+  protected abstract Map<String, Object> getDelegate();
```

Comment:
Please add a Javadoc comment explaining why the keys should still be Strings rather than Keys. (Or change, if they don't.)

Reply:
This requires updating SerDe (type reference and commenting that object mapper should have enum support, considering we have custom value name) or converting map keys to strings on serialization and back on deserialization.

Reply:
Could you please add a comment like `we are mapping from Strings rather than {@link BaseKey}s because ...`?

Reply:
Updated the delegate type to `Map<BaseKey, Object>`